transient-aws/hive-example/cluster_bare_amis.conf

```diff
@@ -0,0 +1,194 @@
+
+#
+# Copyright (c) 2016 Cloudera, Inc. All rights reserved.
+#
+
+#
+# Simple AWS Cloudera Director configuration file with automatic role assignments
+# that works as expected if you use a single instance type for all cluster nodes
+#
+
+#
+# Cluster name
+#
+
+name: ETL-Transient
+
+#
+# Cloud provider configuration (credentials, region or zone and optional default image)
+#
+
+provider {
+    type: aws
+
+    accessKeyId: "REPLACE_ME"
+    secretAccessKey: "REPLACE_ME"
+
+    publishAccessKeys: false
+
+    region: us-west-2[REPLACE_ME]
+
+    subnetId: subnet-41b93525[REPLACE_ME]
+
+    securityGroupsIds: sg-0cf7326a [REPLACE_ME]
+
+    instanceNamePrefix: hive-cdh
+
+    rootVolumeSizeGB: 30 # matching the size of the pre-extracted AMI
+    rootVolumeType: gp2 # OR standard (for EBS magnetic)
+
+    associatePublicIpAddresses: true
+}
+
+#
+# SSH credentials to use to connect to the instances
+#
+
+ssh {
+    username: ec2-user [REPLACE_ME]
+    privateKey: """-----BEGIN RSA PRIVATE KEY-----
+        [REPLACE_ME]
+    -----END RSA PRIVATE KEY-----
+"""
+}
+
+#
+# A list of instance types to use for group of nodes or management services
+#
+
+instances {
+    master {
+        type: c4.xlarge
+        image: ami-775e4f16 [REPLACE_ME]
+        instanceNamePrefix: master-cdh
+        tags {
+            owner: ${?USER}
+        }
+    }
+    worker {
+        type: r3.xlarge
+        image: ami-775e4f16 [REPLACE_ME]
+        instanceNamePrefix: worker-cdh
+        tags {
+            owner: ${?USER}
+        }    
+    }
+
+    worker-spot{
+        type: r3.xlarge
+        image: ami-d2c924b2 [REPLACE_ME]
+        instanceNamePrefix: worker-spot
+        tags {
+            owner: ${?USER}
+        }
+    }
+}
+
+
+#
+# Configuration for Cloudera Manager. Cloudera Director can use an existing instance
+# or bootstrap everything from scratch for a new cluster
+#
+
+cloudera-manager {
+
+    instance: ${instances.master} {
+        tags {
+            application: "Cloudera Manager 5"
+        }
+    }
+
+    #
+    # Automatically activate 60-Day Cloudera Enterprise Trial
+    #
+
+    enableEnterpriseTrial: true
+
+    repository: "http://archive.cloudera.com/cm5/redhat/7/x86_64/cm/5.8/"
```

Comment:
Latest and greatest and in case I am not aware of some optimization that went in.  
I used 5.8 when creating the preloaded AMIs so that will have to match here too.